src/tasks/TaskHelper.ts

```diff
@@ -62,45 +59,56 @@ export interface DockerRunTaskContext extends DockerTaskExecutionContext {
 export type DockerComposeTaskContext = DockerTaskExecutionContext;
 
-export interface TaskHelper {
-    preBuild?(context: DockerBuildTaskContext, buildDefinition: DockerBuildTaskDefinition): Promise<void>;
-    getDockerBuildOptions(context: DockerBuildTaskContext, buildDefinition: DockerBuildTaskDefinition): Promise<DockerBuildOptions>;
-    postBuild?(context: DockerBuildTaskContext, buildDefinition: DockerBuildTaskDefinition): Promise<void>;
-
-    preRun?(context: DockerRunTaskContext, runDefinition: DockerRunTaskDefinition): Promise<void>;
-    getDockerRunOptions(context: DockerRunTaskContext, runDefinition: DockerRunTaskDefinition): Promise<DockerRunOptions>;
-    postRun?(context: DockerRunTaskContext, runDefinition: DockerRunTaskDefinition): Promise<void>;
-}
-
-export function registerTaskProviders(ctx: ExtensionContext): void {
-    const helpers = {
-        netCore: netCoreTaskHelper,
-        node: nodeTaskHelper,
-        python: pythonTaskHelper
-    };
-
-    ctx.subscriptions.push(
-        tasks.registerTaskProvider(
-            'docker-build',
-            new DockerBuildTaskProvider(helpers)
-        )
-    );
-
-    ctx.subscriptions.push(
-        tasks.registerTaskProvider(
-            'docker-run',
-            new DockerRunTaskProvider(helpers)
-        )
-    );
-
-    ctx.subscriptions.push(
-        tasks.registerTaskProvider(
-            'docker-compose',
-            new DockerComposeTaskProvider()
-        )
-    );
+export class TaskHelper {
+    async preBuild(context: DockerBuildTaskContext, buildDefinition: DockerBuildTaskDefinition): Promise<void> {
+        return;
+    }
+
+    async getDockerBuildOptions(context: DockerBuildTaskContext, buildDefinition: DockerBuildTaskDefinition): Promise<DockerBuildOptions> {
+        return buildDefinition.dockerBuild || {};
+    }
+
+    async build(context: DockerBuildTaskContext, buildDefinition: DockerBuildTaskDefinition) {
+        const client = await ext.runtimeManager.getClient();
+
+        const options = buildDefinition.dockerBuild;
+        const command = await client.buildImage({
+            pull: options.pull,
+            file: options.dockerfile,
+            args: options.buildArgs,
+            labels: getAggregateLabels(options.labels, defaultVsCodeLabels),
+            tags: [options.tag],
+            stage: options.target,
+            platform: options.platform,
+            customOptions: options.customOptions,
+            path: options.context,
+        });
+
+        const runner = context.terminal.getCommandRunner({
+            folder: context.folder,
+            token: context.cancellationToken,
+        });
+
+        await runner(command);
+    }
+
+    async postBuild(context: DockerBuildTaskContext, buildDefinition: DockerBuildTaskDefinition): Promise<void> {
+        return;
+    }
+
+    async preRun(context: DockerRunTaskContext, runDefinition: DockerRunTaskDefinition): Promise<void> {
+        return;
+    }

```

Comment:
More missing new lines

Reply:
:disappointed_relieved: I should've looked closer